### PR TITLE
[FIX] mail: prevent timed-out error when quit pop server

### PR DESCRIPTION
**Pull request by omra-odoo**
OSError: cannot read from timed out object while evaluating a
"model._fetch_mails()", it occurs because when we try to quit a POP server
of incoming mail using the "pop_server.quit()" method, sometimes it is not
able to reach the remote server to quit the pop server, so it throws an OSError.

see https://drive.google.com/file/d/1Xh4gnS5xUSB4e_XOshyC7r5814gEXSdR/view?usp=sharing

We added "pop_server.quit()" to the try block; if it fails to quit the server,
a warning will be displayed using the logger.

sentry-3932133536

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by omra-odoo** [comment]
Hello! @abz-odoo @adwid @yelizariev
Can you please take a look at the PR?
Thanks!

File: addons/mail/models/fetchmail.py [hunk 1]
```diff
@@ -250,7 +250,10 @@ def fetch_mail(self):
                     _logger.info("General failure when trying to fetch mail from %s server %s.", server.server_type, server.name, exc_info=True)
                 finally:
                     if pop_server:
-                        pop_server.quit()
+                        try:
+                            pop_server.quit()
+                        except Exception:
```
Comment (yelizariev) [inline addons/mail/models/fetchmail.py:255]:
```suggestion
                        except OSError:
```

File: addons/mail/models/fetchmail.py [hunk 2]
```diff
@@ -250,7 +250,10 @@ def fetch_mail(self):
...
+                            _logger.info('Cannot Reach to the server %s.', server.name, exc_info=True)
```
Comment (yelizariev) [inline addons/mail/models/fetchmail.py:256]:
```suggestion
                            _logger.warning('Enable to properly finish pop connection: %s.', server.name, exc_info=True)
```

**Review by yelizariev — COMMENTED** [review]
Target to master instead.

Also, update commit message

https://www.odoo.com/documentation/16.0/contributing/development/git_guidelines.html#commit-message-header

image

**Comment by omra-odoo** [comment]
@yelizariev Changes are done

**Comment by omra-odoo** [comment]
@odony Changes are done.

Thanks.

**Comment by yelizariev** [comment]
> why it cannot be done in stable (e.g. 16)?

At first glance It looked for me as a logs cleaning and not really a fix. But it actually makes sure that all other mail servers are processed. Let's do it to v16 then

**Comment by yelizariev** [comment]
@omra-odoo please target to v16

**Comment by omra-odoo** [comment]
@yelizariev  I have changed the target to v16.

Thanks.

**Review by yelizariev — APPROVED** [review]

**Review by tde-banana-odoo — APPROVED** [review]

**Comment by kitan191** [comment]
@robodoo r+

**Pull request closed without merging**